### PR TITLE
Revert "Optimization on running prePreEnqueuePlugins before adding pods into activeQ"

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This reverts commit c01fa8279d3af2896744b2d6e011786d2bd62e68.

The discussion: https://github.com/kubernetes/kubernetes/pull/115583#issuecomment-1502637388

---

This PR fixes that the PreEnqueue plugins aren't executed if Pods proceed to activeQ through backoffQ.

It doesn't affect the vanilla kubernetes because if the scheduling gate plugin once allow Pods to go to activeQ, it always allows them afterward. But, it may break custom PreEnqueue plugins.

This PR also adds the description of this issue as the known issue on the release note of v1.27.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fix a 1.27 regression that custom PreEnqueue plugins aren't executed for Pods proceeding to activeQ through backoffQ.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by Huang-Wei — COMMENTED** [review]
/approve

We will see if we can make it to 1.27, or we can wait for the first 1.27 patch release.

**Comment by liggitt** [comment]
That seems like a good candidate for inclusion in a 1.27.1 patch release, and noting the issue as a known issue for someone using custom scheduler plug-ins in 1.27.0.

**Comment by sanposhiho** [comment]
ok, I'll include the change to add the description as the known issue to the release note once v1.27 is released.

**Comment by alculquicondor** [comment]
> as a follow-up, I'd recommend scheduling folks open an issue to track improving the testing of external plugins in a way that would have prevented this in the first place

@sanposhiho can you add an integration test?

**Comment by Huang-Wei** [comment]
> as a follow-up, I'd recommend scheduling folks open an issue to track improving the testing of external plugins in a way that would have prevented this in the first place

Creating https://github.com/kubernetes/kubernetes/issues/117215 to track follow-ups.

**Comment by Huang-Wei** [comment]
> @sanposhiho can you add an integration test?

We can have a separate test. And have this PR easy to back-port.

**Comment by sanposhiho** [comment]
added the known issue section on the release note v1.26.

**Comment by sanposhiho** [comment]
> can you add an integration test?

> We can have a separate test. And have this PR easy to back-port.

Sure. I'll follow up this in another PR.

**Comment by cici37** [comment]
/unhold
Sorry accidentally did it in the wrong pr.

File: CHANGELOG/CHANGELOG-1.27.md [hunk 1]
```diff
@@ -198,6 +200,15 @@ name | architectures
 
 ## Changelog since v1.26.0
 
+## Known Issues
+
+### The PreEnqueue extension point doesn't work for Pods going to activeQ through backoffQ
+
+In v1.26.0, we've found the bug that the PreEnqueue extension point doesn't work for Pods going to activeQ through backoffQ.
```
Comment (alculquicondor) [inline CHANGELOG/CHANGELOG-1.27.md:207]: 1.27.0

Comment (sanposhiho) [reply CHANGELOG/CHANGELOG-1.27.md:207]: oops, I'll fix in another PR 🙏

**Pull request merged**